mne/preprocessing/ica.py

```diff
@@ -2017,6 +2021,27 @@ def save(self, fname, *, overwrite=False, verbose=None):
             _write_ica(fid, self)
         return self
 
+    @property
+    def explained_variance_ratio_(self):
+        """The relative variance explained by the ICA components.
+
+        Returns
+        -------
+        array, shape (n_ica_components,) | None
+            Relative explained variance for each ICA component. If ICA has
+            not been fit, returns ``None``.
+        """
+        if self.current_fit == 'unfitted':
+            logger.info(
+                'Explained variance ratio is None (model not yet fitted)'
+            )
+            return None
+
+        relative_explained_var = (
+            self.pca_explained_variance_ / self.pca_explained_variance_.sum()
```

Comment:
are you sure it's correct if you don't compute all the pca components and just a subset? this number should not depend on the number of PCA components you compute.

Reply:
We always keep **all** PCA components, and their respective explained variances are in `ICA.pca_explained_variance_`. Since this array contains the variances for all PCA components, it's okay to normalize / unitize it.

Of these PCA components, only a subset is being sent to ICA. What I'm doing is picking the variances explained by this subset and returning them.

Does that make sense to you?

Reply:
The only place where we allow "dropping" PCA components since version 0.24 or so is when running `ICA.apply()`, where users may pass `n_pca_components` to select how many components to use for unmixing

Reply:
@larsoner Could you share your thoughts please? I'm 99% sure I'm doing the right thing here, but … you never know!

Reply:
wait a minute this is PCA components explained variance not ICA explained variance !

also ICA components are not orthogonal so the notion of explained variance for a group of components is fishy. At best you can see how much variance you keep by including only component but since all components are not orthogonal to each other the sum will not be 1.

Danger zone...